test/port-finder-socket.test.js

```diff
@@ -88,60 +86,126 @@ describe('portfinder', function () {
+    ])(`the %s method (promise: %p)`, function (name, isPromise, method) {
       test("with a directory that doesn't exist should respond with the first free socket (test.sock)", function (done) {
-        portfinder.getSocket({
-          path: path.join(badDir, 'test.sock'),
-        }, function (err, socket) {
-          expect(err).toBeNull();
-          expect(socket).toEqual(path.join(badDir, 'test.sock'));
-          done();
-        });
+        if (isPromise) {
+          method({
+            path: path.join(badDir, 'test.sock'),
+          })
+            .then(function (socket) {
+              expect(socket).toEqual(path.join(badDir, 'test.sock'));
+              done();
+            })
+            .catch(function (err) {
+              done(err);
+            });
+        } else {
+          method({
+            path: path.join(badDir, 'test.sock'),
+          }, function (err, socket) {
+            if (err) {
+              done(err);
+              return;
+            }
+            expect(err).toBeNull();
+            expect(socket).toEqual(path.join(badDir, 'test.sock'));
+            done();
+          });
+        }
       });
 
       test("with a nested directory that doesn't exist should respond with the first free socket (test.sock)", function (done) {
-        portfinder.getSocket({
-          path: path.join(badDir, 'deeply', 'nested', 'test.sock'),
-        }, function (err, socket) {
-          expect(err).toBeNull();
-          expect(socket).toEqual(path.join(badDir, 'deeply', 'nested', 'test.sock'));
-          done();
-        });
+        if (isPromise) {
+          method({
+            path: path.join(badDir, 'deeply', 'nested', 'test.sock'),
+          })
+            .then(function (socket) {
+              expect(socket).toEqual(path.join(badDir, 'deeply', 'nested', 'test.sock'));
+              done();
+            })
+            .catch(function (err) {
+              done(err);
+            });
+        } else {
+          method({
+            path: path.join(badDir, 'deeply', 'nested', 'test.sock'),
+          }, function (err, socket) {
+            expect(err).toBeNull();
+            expect(socket).toEqual(path.join(badDir, 'deeply', 'nested', 'test.sock'));
+            done();
+          });
+        }
       });
 
-      test('with a directory that exists should respond with the first free socket (test.sock)', function (done) {
-        portfinder.getSocket({
-          path: path.join(socketDir, 'exists.sock'),
-        }, function (err, socket) {
-          expect(err).toBeNull();
-          expect(socket).toEqual(path.join(socketDir, 'exists.sock'));
-          done();
-        });
+      // We don't use `test.sock` here due to some race condition on Windows in freeing the `test.sock` file
+      // when we close the servers.
+      test('with a directory that exists should respond with the first free socket (foo.sock)', function (done) {
```

Comment:
The test was mentioning `test.sock`, but the code was using `exists.sock`. I'm guessing this was due to an issue on windows where the socket files were not being cleared as part of `cleanup`. While this was fixed, I think there's still some weird race condition where calling this function too quickly will still fail on Windows if using `test.sock`.

As such, I went with calling it `foo.sock` to avoid the name collision and that I renamed it as I didn't like calling it `exists.sock` as that felt like it was implying the socket file already existed.

Reply:
okay cool - might answer why were now calling `fs.unlinkSync(socket);` as well it seems

thanks for the note